src/analysis/symbols.rs

```diff
@@ -0,0 +1,147 @@
+        }
+
+        for typ in ns.types.iter().map(|t| t.as_ref().unwrap()) {
+            let symbol = Symbol {
+                crate_name: crate_name.cloned(),
+                name: typ.get_name(),
+                ..Default::default()
+            };
+            
+            match *typ {
+                Type::Alias(Alias { ref c_identifier, .. }) => {
+                    info.insert(c_identifier, symbol);
+                }
+                Type::Enumeration(Enumeration {
+                    ref name,
+                    ref c_type,
+                    ref members,
+                    ref functions,
+                        ..
+                }) |
+                    Type::Bitfield(Bitfield {
+                        ref name,
+                        ref c_type,
+                        ref members,
+                        ref functions,
+                        ..
+                    }) => {
+                    info.insert(c_type, symbol);
+                    for member in members {
+                        let symbol = Symbol {
+                            crate_name: crate_name.cloned(),
+                            owner_name: Some(name.clone()),
+                            name: member.name.to_camel(),
+                        };
+                        info.insert(&member.c_identifier, symbol);
+                    }
+                    for func in functions {
+                        let symbol = Symbol {
+                            crate_name: crate_name.cloned(),
+                            owner_name: Some(name.clone()),
+                            name: func.name.clone(),
+                        };
+                        info.insert(func.c_identifier.as_ref().unwrap(), symbol);
+                    }
+                }
+                Type::Record(Record { ref name, ref c_type, ref functions, .. }) |
+                    Type::Class(Class { ref name, ref c_type, ref functions, .. }) |
+                    Type::Interface(Interface { ref name, ref c_type, ref functions, .. }) => {
+                    info.insert(c_type, symbol);
+                    for func in functions {
+                        let symbol = Symbol {
+                            crate_name: crate_name.cloned(),
+                            owner_name: Some(name.clone()),
+                            name: func.name.clone(),
+                        };
+                        info.insert(func.c_identifier.as_ref().unwrap(), symbol);
+                    }
+                }
+                _ => { }
+            }
+        }
+    }
+
+    info
+}
+
+impl Info {
+    pub fn by_c_name(&self, name: &str) -> Option<&Symbol> {
+        self.c_name_index.get(name).map(|&id| &self.symbols[id as usize])
+    }
+
+    pub fn by_c_name_mut(&mut self, name: &str) -> Option<&mut Symbol> {
+        if let Some(&id) = self.c_name_index.get(name) {
+            Some(&mut self.symbols[id as usize])
+        }
+        else {
```

Comment:
A large part of gir still uses the above style. I guess I never noticed the discrepancy. I might evaluate rustfmt again to avoid this kind of issues :)